File: Timeline.Web/Models/Audit.cs
using System;

namespace Timeline.Models
{
    public class Audit
    {
        public Guid Id { get; set; }
        public DateTime TimeStamp { get; set; }
        public AuditAction Action { get; set; }
        public string Log { get; set; }
        public AuditOrigin Origin { get; set; }
        public User User { get; set; }
        public Team Team { get; set; }
    }

    public enum AuditOrigin
    {
        JobController,
        TeamsController,
        UserController
    }

    public enum AuditAction
    {
        UserJoinedTeam,
        UserLeftTeam,
    }
}

File: Timeline.Neo/ClientApp/src/app/_models/audit.ts
import {User} from "./user";
import {Team} from "./team";

enum AuditAction
{
}

enum AuditOrigin
{
}

export interface Audit {
  id?: string;
  timeStamp: Date;
  action: AuditAction;
  log: string;
  origin: AuditOrigin;
  user: User;
  team: Team;
}
